pull/github.go

```diff
@@ -860,6 +860,7 @@ type v4Commit struct {
 			OID string
 		}
 	} `graphql:"parents(first: 3)"`
+	Signature *v4GitSignature
```

Comment:
Should we be looking for a https://docs.github.com/en/graphql/reference/objects#gpgsignature or a https://docs.github.com/en/graphql/reference/objects#smimesignature ? It seems like we're using the later?

Reply:
I was going off the generic interface that they both implement https://docs.github.com/en/graphql/reference/interfaces#gitsignature

Perhaps it makes more sense to key off of the implemented type and grab the extra fields

Reply:
I refactored to inline fragment the implementation objects and expose it that way